client/pdo/client/commands/collection.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import copy
+import logging
+import os
+import toml
+import typing
+
+from zipfile import ZipFile
+
+import pdo.client.builder.shell as pshell
+import pdo.client.builder.script as pscript
+import pdo.client.builder as pbuilder
+from pdo.common.utility import experimental
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    'export_contract_collection',
+    'import_contract_collection',
+    'script_command_export',
+    'script_command_import',
+    'do_collection',
+    'load_commands',
+]
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+def __find_contracts__(context : dict) -> typing.List[str] :
+    """Find all contract save files in a context dictionary
+
+    @type context : dict
+    @param context : the export context
+    """
+    save_files = []
+    for k, v in context.items() :
+        if k == 'save_file' :
+            save_files.append(v)
```

Comment:
Here you are making assumptions about the schema of the context file (by looking for key word 'save_file'). My concern is that the pdo repo AFAIK, does not have any examples of context files. The context files are only introduced in the pdo contracts repo (may be I am wrong?). If this is the case, makes me wonder why we wouldn't maintain some of the client/commands (such as collection, and even context.py) as part of the pdo/contacts repo? Thanks!

Reply:
This is pretty well documented in the issue so i don't feel a particular need for examples. There is a need for a more thorough test that involves multiple users with completely distinct key sets. We do not currently have the infrastructure to do that. I would suggest that the right way to address this is to add an issue on multi user tests.